### PR TITLE
ensure flexion is correct in german_noun_analysis

### PR DESCRIPTION
**Pull request by lsmith77**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1857,37 +1858,48 @@ def german_noun_analysis(word, genus_only=False):
...
-        while i < len(word) - 4:
-            partial_word = word[i:].capitalize()
-            i += 1
 
-            result = german_nouns[partial_word]
+        # skip the last 2 letters
+        while i < len(word) - 2:
+            partial_word = word[i:]
+
+            # avoid cases like 'Ende' at the end of 'Arbeitgebende'
```
Comment (lsmith77) [inline app/main.py]: @tje-ww any advice here? without this, `Arbeitgebende` will be reduced to `Ende` which leads to determining the wrong genus. are there other such cases you are aware of?

File: app/main.py [hunk 2]
```diff
@@ -1840,59 +1842,87 @@ def word_types_overlap(a_word_types, b_word_types):
...
+
+    if "genus" in result:
+        return result
+
+    if "genus 1" in result:
+        result["genus"] = result["genus 1"]
+
+        return result
+
+    if word[-5:].lower() == "leute":
```
Comment (lsmith77) [inline app/main.py:1986]: edge case fix

File: app/main.py [hunk 3]
```diff
@@ -1840,59 +1842,87 @@ def word_types_overlap(a_word_types, b_word_types):
...
-            if result != None:
-                return result
+        result = german_noun_lookup(partial_word.capitalize())
+        if result == None:
+            i += 1
+            continue
 
-    if isinstance(result, list) and "genus 1" in result:
-        result["genus"] = result["genus 1"]
+        result["Lemma"] = word
```
Comment (lsmith77) [inline app/main.py]: ensure we don't return the data for the partial result

File: app/rules.py [hunk 4]
```diff
@@ -424,6 +424,39 @@
     "aber",
 ]
 
+rules["de"]["gender_neutral_nouns"] = {
+    "Ierende": {
```
Comment (lsmith77) [inline app/rules.py:427]: this is "hack" to handle gender-neutral words like "Studierende", "Rotierende", "Levitierende" etc.

File: app/rules.py [hunk 5]
```diff
@@ -424,6 +424,39 @@
...
+            "dativ singular": "Ierender",
+            "dativ plural": "Ierenden",
+            "akkusativ singular": "Ierende",
+            "akkusativ plural": "Ierende",
+        },
+        "lemma": "Ierende",
+        "pos": ["Substantiv", "adjektivische Deklination"],
+        "genus": "f",
+    },
+    "Gebende": {
```
Comment (lsmith77) [inline app/rules.py:442]: handle cases like "Arbeitgebende"

**Review by lnazarenko — APPROVED** [review]

**Pull request merged**